web/src/components/navbar/NavBarPopupMenu.test.tsx

```diff
@@ -0,0 +1,281 @@
+import { NavBarPopupMenu, Props as NavBarPopupMenuProps } from "@/components/navbar/NavBarPopupMenu";
+import { getMergedConfig } from "@/contexts/configContext";
+import * as api from "@/lib/api-client/apiClient";
+import { IsAuthenticated } from "@/lib/auth/AuthContext";
+import { triggerSignIn } from "@/lib/auth/sessionHelper";
+import { onSignOut } from "@/lib/auth/signinHelper";
+import { ROUTES } from "@/lib/domain-logic/routes";
+import { withAuth } from "@/test/helpers/helpers-renderers";
+import { mockPush, useMockRouter } from "@/test/mock/mockRouter";
+import { useMockUserData } from "@/test/mock/mockUseUserData";
+import { setupStatefulUserDataContext, WithStatefulUserData } from "@/test/mock/withStatefulUserData";
+import {
+  BusinessUser,
+  generateProfileData,
+  generateUser,
+  generateUserData,
+  UserData,
+} from "@businessnjgovnavigator/shared";
+import { fireEvent, render, screen, waitFor } from "@testing-library/react";
+
+const Config = getMergedConfig();
+
+const mockApi = api as jest.Mocked<typeof api>;
+
+jest.mock("next/router", () => ({ useRouter: jest.fn() }));
+jest.mock("@/lib/data-hooks/useUserData", () => ({ useUserData: jest.fn() }));
+jest.mock("@/lib/data-hooks/useRoadmap", () => ({ useRoadmap: jest.fn() }));
+jest.mock("@/lib/api-client/apiClient", () => ({ postSelfReg: jest.fn() }));
+jest.mock("@/lib/auth/signinHelper", () => {
+  const originalSigninHelper = jest.requireActual("@/lib/auth/signinHelper");
+  return {
+    ...originalSigninHelper,
+    onSignOut: jest.fn(),
+  };
+});
+jest.mock("@/lib/auth/sessionHelper", () => ({
+  triggerSignIn: jest.fn(),
+}));
```

Comment:
Also might write a follow up because this is a bit weird how we have triggerSignIn and triggerSignOut and onSignOut and onSignIn and we're mixing them here which is also what we're doing in main I believe. Something to investigate and potentially refactor maybe?